File: problems/01/second/main.py
import sys


def main():
    file = sys.argv[1]
    with open(file) as f:
        lefts = []
        rights = {}

        for line in f.readlines():
            [left_raw, right_raw] = line.split()
            lefts.append(int(left_raw.strip()))

            right = int(right_raw.strip())
            if rights.get(right):
                rights[right] += 1
            else:
                rights[right] = 1

        similarity = 0
        for left in lefts:
            similarity += rights.get(left, 0) * left

        print(similarity)


if __name__ == "__main__":
    main()

File: problems/01/second/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file := os.Args[1]

	bcontent, err := os.ReadFile(file)
	if err != nil {
		panic("Failed to open file")
	}

	scontent := string(bcontent)

	lines := strings.Split(scontent, "\n")
	nlines := len(lines)
	lefts := make([]int, nlines)
	freqs := make(map[int]int)

	for i, line := range lines {
		parts := strings.Split(line, "   ")

		lefts[i], err = strconv.Atoi(parts[0])
		if err != nil {
			panic("bad left value")
		}
		right, err := strconv.Atoi(parts[1])
		if err != nil {
			panic("bad right value")
		}

		freqs[right] += 1
	}

	total := 0
	for _, left := range lefts {
		total += left * freqs[left]
	}

	fmt.Println(total)
}
